pkg/domain/infra/abi/play.go

```diff
@@ -964,3 +978,36 @@ func (ic *ContainerEngine) PlayKubeDown(ctx context.Context, body io.Reader, _ e
 
 	return reports, nil
 }
+
+// playKubeSecret allows users to create and store a kubernetes secret as a podman secret
+func (ic *ContainerEngine) playKubeSecret(secret *v1.Secret) (*entities.SecretCreateReport, error) {
+	r := &entities.SecretCreateReport{}
+
+	// Create the secret manager before hand
+	secretsManager, err := ic.Libpod.SecretsManager()
+	if err != nil {
+		return nil, err
+	}
+
+	// only store the data!!! if we store the entire secret marshaled,
+	// we cannot unmarshal it later. (yaml.unmarshal does not like string --> byte conversion)
+	data, err := yaml.Marshal(secret)
+	if err != nil {
+		return nil, err
+	}
+
+	secretsPath := ic.Libpod.GetSecretsStorageDir()
+	opts := make(map[string]string)
+	opts["path"] = filepath.Join(secretsPath, "filedriver")
+	// maybe k8sName(data)...
+	// using this does not allow the user to use the name given to the secret
+	// but keeping secret.Name as the ID can lead to a collision.
+	secretID, err := secretsManager.Store(secret.Name, data, "file", opts)
```

Comment:
I thought we'd use `k8sName(data)` as the name?

Reply:
@vrothberg lets talk at watercooler, I think using `k8sName` breaks the separate file usage of secrets since the user will not know the name given. I can change the function back to just being `serviceContainerName` but thought the change made the function more applicable to other types in the future.

Reply:
Added a comment below i think `secret.Name` is fine and on collision secret must be updated but if support for `immutable` is added then on collision it should error out for specific secrets where `immutable: true` is set. See below comment for more details.

Reply:
Great input, @flouthoc! Thank you